codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Exception.java

```diff
@@ -47,6 +63,13 @@ public Http2Error error() {
     }
 
     /**
+     * If a connection error should attempt a graceful shutdown.
+     * @return {@code true} if a connection error should attempt a graceful shutdown. {@code false} otherwise.
+     */
+    public boolean gracefulShutdown() {
```

Comment:
I'm a bit hesitant to introduce an enum because it conflicts a bit with the `isStreamError` logic. If we introduce this enum its almost like we should transition over to using this enum instead of checking the type of exception, but then there are already some semantics associated with the protocol and other assumptions we make based upon types.

Basically what I want to capture in the method name is something like the following: `ifAShutdownIsWarrantedThenDoAGracefulShutdown`. So it is clear the logic to execute a shutdown is not necessarily made at time time of exception creation. Maybe `shouldShutdownBeGraceful`?